src/Workspaces/Core/MSBuild.BuildHost/Rpc/JsonSettings.cs

```diff
@@ -2,18 +2,22 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Text;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Serialization;
 
 namespace Microsoft.CodeAnalysis.MSBuild.Rpc;
 
 internal static class JsonSettings
 {
+    public static readonly Encoding StreamEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+
     public static readonly JsonSerializerSettings SingleLineSerializerSettings = new JsonSerializerSettings
     {
         Formatting = Newtonsoft.Json.Formatting.None,
         NullValueHandling = NullValueHandling.Ignore,
         ContractResolver = new CamelCasePropertyNamesContractResolver(),
         TypeNameHandling = TypeNameHandling.None,
+        StringEscapeHandling = StringEscapeHandling.EscapeNonAscii
```

Comment:
why?

Reply:
Comment added: we had a failure in CI on Friday where the stdin/out of the external process was impacted by the process codepage so the Unicode BOM got mangled. Curiously not stable! Either way ensuring that the actual text on the stream stays out of any potential complexity ensures we aren't broken by obscure Windows settings that way.